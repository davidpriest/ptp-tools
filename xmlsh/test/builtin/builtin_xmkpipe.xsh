# Test of mkpipe

# Test stream pipes

xmkpipe s

{ while read a  ; do echo $a ; xtype $a ; done <(s) ; echo Done ; } &

for i in <[ 1 to 100 ]> ; do 
   echo <[ <elem>{$i}</elem> ]>
done  >(s)
xmkpipe -close s
wait
echo Text Complete


xmkpipe -xml x

{ while xread a  ; do xecho $a ; xtype $a ; done <(x) ; echo Done ;  } &

for i in <[ 1 to 100 ]> ; do 
   xecho <[ <elem>{$i}</elem> ]> 
done  >(x)
xmkpipe -close x
wait
echo XML Complete