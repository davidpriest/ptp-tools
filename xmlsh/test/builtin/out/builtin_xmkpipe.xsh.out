<elem>1</elem>
xs:string
<elem>2</elem>
xs:string
<elem>3</elem>
xs:string
<elem>4</elem>
xs:string
<elem>5</elem>
xs:string
<elem>6</elem>
xs:string
<elem>7</elem>
xs:string
<elem>8</elem>
xs:string
<elem>9</elem>
xs:string
<elem>10</elem>
xs:string
<elem>11</elem>
xs:string
<elem>12</elem>
xs:string
<elem>13</elem>
xs:string
<elem>14</elem>
xs:string
<elem>15</elem>
xs:string
<elem>16</elem>
xs:string
<elem>17</elem>
xs:string
<elem>18</elem>
xs:string
<elem>19</elem>
xs:string
<elem>20</elem>
xs:string
<elem>21</elem>
xs:string
<elem>22</elem>
xs:string
<elem>23</elem>
xs:string
<elem>24</elem>
xs:string
<elem>25</elem>
xs:string
<elem>26</elem>
xs:string
<elem>27</elem>
xs:string
<elem>28</elem>
xs:string
<elem>29</elem>
xs:string
<elem>30</elem>
xs:string
<elem>31</elem>
xs:string
<elem>32</elem>
xs:string
<elem>33</elem>
xs:string
<elem>34</elem>
xs:string
<elem>35</elem>
xs:string
<elem>36</elem>
xs:string
<elem>37</elem>
xs:string
<elem>38</elem>
xs:string
<elem>39</elem>
xs:string
<elem>40</elem>
xs:string
<elem>41</elem>
xs:string
<elem>42</elem>
xs:string
<elem>43</elem>
xs:string
<elem>44</elem>
xs:string
<elem>45</elem>
xs:string
<elem>46</elem>
xs:string
<elem>47</elem>
xs:string
<elem>48</elem>
xs:string
<elem>49</elem>
xs:string
<elem>50</elem>
xs:string
<elem>51</elem>
xs:string
<elem>52</elem>
xs:string
<elem>53</elem>
xs:string
<elem>54</elem>
xs:string
<elem>55</elem>
xs:string
<elem>56</elem>
xs:string
<elem>57</elem>
xs:string
<elem>58</elem>
xs:string
<elem>59</elem>
xs:string
<elem>60</elem>
xs:string
<elem>61</elem>
xs:string
<elem>62</elem>
xs:string
<elem>63</elem>
xs:string
<elem>64</elem>
xs:string
<elem>65</elem>
xs:string
<elem>66</elem>
xs:string
<elem>67</elem>
xs:string
<elem>68</elem>
xs:string
<elem>69</elem>
xs:string
<elem>70</elem>
xs:string
<elem>71</elem>
xs:string
<elem>72</elem>
xs:string
<elem>73</elem>
xs:string
<elem>74</elem>
xs:string
<elem>75</elem>
xs:string
<elem>76</elem>
xs:string
<elem>77</elem>
xs:string
<elem>78</elem>
xs:string
<elem>79</elem>
xs:string
<elem>80</elem>
xs:string
<elem>81</elem>
xs:string
<elem>82</elem>
xs:string
<elem>83</elem>
xs:string
<elem>84</elem>
xs:string
<elem>85</elem>
xs:string
<elem>86</elem>
xs:string
<elem>87</elem>
xs:string
<elem>88</elem>
xs:string
<elem>89</elem>
xs:string
<elem>90</elem>
xs:string
<elem>91</elem>
xs:string
<elem>92</elem>
xs:string
<elem>93</elem>
xs:string
<elem>94</elem>
xs:string
<elem>95</elem>
xs:string
<elem>96</elem>
xs:string
<elem>97</elem>
xs:string
<elem>98</elem>
xs:string
<elem>99</elem>
xs:string
<elem>100</elem>
xs:string
Done
Text Complete
<elem>1</elem>
element()
<elem>2</elem>
element()
<elem>3</elem>
element()
<elem>4</elem>
element()
<elem>5</elem>
element()
<elem>6</elem>
element()
<elem>7</elem>
element()
<elem>8</elem>
element()
<elem>9</elem>
element()
<elem>10</elem>
element()
<elem>11</elem>
element()
<elem>12</elem>
element()
<elem>13</elem>
element()
<elem>14</elem>
element()
<elem>15</elem>
element()
<elem>16</elem>
element()
<elem>17</elem>
element()
<elem>18</elem>
element()
<elem>19</elem>
element()
<elem>20</elem>
element()
<elem>21</elem>
element()
<elem>22</elem>
element()
<elem>23</elem>
element()
<elem>24</elem>
element()
<elem>25</elem>
element()
<elem>26</elem>
element()
<elem>27</elem>
element()
<elem>28</elem>
element()
<elem>29</elem>
element()
<elem>30</elem>
element()
<elem>31</elem>
element()
<elem>32</elem>
element()
<elem>33</elem>
element()
<elem>34</elem>
element()
<elem>35</elem>
element()
<elem>36</elem>
element()
<elem>37</elem>
element()
<elem>38</elem>
element()
<elem>39</elem>
element()
<elem>40</elem>
element()
<elem>41</elem>
element()
<elem>42</elem>
element()
<elem>43</elem>
element()
<elem>44</elem>
element()
<elem>45</elem>
element()
<elem>46</elem>
element()
<elem>47</elem>
element()
<elem>48</elem>
element()
<elem>49</elem>
element()
<elem>50</elem>
element()
<elem>51</elem>
element()
<elem>52</elem>
element()
<elem>53</elem>
element()
<elem>54</elem>
element()
<elem>55</elem>
element()
<elem>56</elem>
element()
<elem>57</elem>
element()
<elem>58</elem>
element()
<elem>59</elem>
element()
<elem>60</elem>
element()
<elem>61</elem>
element()
<elem>62</elem>
element()
<elem>63</elem>
element()
<elem>64</elem>
element()
<elem>65</elem>
element()
<elem>66</elem>
element()
<elem>67</elem>
element()
<elem>68</elem>
element()
<elem>69</elem>
element()
<elem>70</elem>
element()
<elem>71</elem>
element()
<elem>72</elem>
element()
<elem>73</elem>
element()
<elem>74</elem>
element()
<elem>75</elem>
element()
<elem>76</elem>
element()
<elem>77</elem>
element()
<elem>78</elem>
element()
<elem>79</elem>
element()
<elem>80</elem>
element()
<elem>81</elem>
element()
<elem>82</elem>
element()
<elem>83</elem>
element()
<elem>84</elem>
element()
<elem>85</elem>
element()
<elem>86</elem>
element()
<elem>87</elem>
element()
<elem>88</elem>
element()
<elem>89</elem>
element()
<elem>90</elem>
element()
<elem>91</elem>
element()
<elem>92</elem>
element()
<elem>93</elem>
element()
<elem>94</elem>
element()
<elem>95</elem>
element()
<elem>96</elem>
element()
<elem>97</elem>
element()
<elem>98</elem>
element()
<elem>99</elem>
element()
<elem>100</elem>
element()
Done
XML Complete
